### ISSUE TITLE
improve: Complexity Analysis

### ISSUE THREAD

**Issue by Utkarsh1504** [Utkarsh1504/DSA-Java#46]
### ⭐ Summary

Please improve this lesson and give the following explanation:
- Memory in Computer Science
- What is Complexity Analysis
- why complexity analysis
- Complexity analysis of an algorithm
- types of complexities
- example

### 💻 Location

Location is at

section: 7
lesson: A
order: 7A
filename: [intro-complexity.md](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/intro-complexity.md)

### ❌ Additional Info

You can take the help of other markdown files in the [/lesson] directory so that you get the idea of in which fashion you should write.

If you don't know how to do markdown, don't worry it is quite easy, just follow this docs-https://guides.github.com/features/mastering-markdown/

If your concepts are not that good follow this Time Complexity video of @kunal-kushwaha (https://www.youtube.com/watch?v=mV3wrLBbuuE)

**Comment by Abhijit25Mishra** [comment]
Yep, I can take this up, I just finished studying this in my sem 3.

**Comment by rushikeshadhav** [comment]
Can I contribute?
Please assign this to me.
Thank you

**Comment by Utkarsh1504** [comment]
@Abhijit25Mishra @rushikeshadhav we want to add the notes of DSA-Bootcamp-Java as Kunal taught on his Youtube channel.
If anyone can add his notes then I'll assign this issue to him.

**Comment by Abhijit25Mishra** [comment]
yeah sure i will add his notes.

**Comment by Utkarsh1504** [comment]
Ok sure go ahead 👍

**Comment by rushikeshadhav** [comment]
Yes, sure I will do it

**Comment by Utkarsh1504** [comment]
@rushikeshadhav I have assigned you an issue already, please work on that, this has been assigned to @Abhijit25Mishra.
I will add more issue regarding complexity and will assign you!

**Comment by sbaijal** [comment]
Can I work on this issue?
I am in my 5th semester and quite clear with DSA concepts.

**Comment by Abhijit25Mishra** [comment]
> Can I work on this issue?
> I am in my 5th semester and quite clear with DSA concepts.

While I understand where ur coming from , please do give the new comers a chance. If I face any problems along the way I will definitely contact you for guidance. Thanks 😊

**Comment by rushikeshadhav** [comment]
@Utkarsh1504  Hey, I don't know how to or where to make the changes. Could you please guide me. I ran the "npm install --global gatsby-cli" command but it is giving error. I have no idea how can I add Complexity analysis.

**Comment by Utkarsh1504** [comment]
@rushikeshadhav this issue is not assigned to you, please work only on those assigned to you. Please read the issue, the file is already associated there in which you have to do contribution. And one thing before running `npm install` make sure you have node installed on your system
